app/src/main/java/com/kabouzeid/gramophone/appwidgets/AppWidgetCard.java

```diff
@@ -0,0 +1,228 @@
+        final int[] mAppWidgetIds = appWidgetManager.getAppWidgetIds(new ComponentName(context,
+                getClass()));
+        return mAppWidgetIds.length > 0;
+    }
+
+    /**
+     * Handle a change notification coming over from
+     * {@link MusicService}
+     */
+    public void notifyChange(final MusicService service, final String what) {
+        if (hasInstances(service)) {
+            if (MusicService.META_CHANGED.equals(what) || MusicService.PLAY_STATE_CHANGED.equals(what)) {
+                performUpdate(service, null);
+            }
+        }
+    }
+
+    /**
+     * Update all active widget instances by pushing changes
+     */
+    public void performUpdate(final MusicService service, final int[] appWidgetIds) {
+        final RemoteViews appWidgetView = new RemoteViews(service.getPackageName(), R.layout.app_widget_card);
+
+        final boolean isPlaying = service.isPlaying();
+        final Song song = service.getCurrentSong();
+
+        // Set the titles and artwork
+        if (TextUtils.isEmpty(song.title) && TextUtils.isEmpty(song.artistName)) {
+            appWidgetView.setViewVisibility(R.id.media_titles, View.INVISIBLE);
+        } else {
+            if (TextUtils.isEmpty(song.artistName) || TextUtils.isEmpty(song.albumName)) {
+                appWidgetView.setTextViewText(R.id.text_separator, "");
+            } else {
+                appWidgetView.setTextViewText(R.id.text_separator, "•");
+            }
+
+            appWidgetView.setViewVisibility(R.id.media_titles, View.VISIBLE);
+            appWidgetView.setTextViewText(R.id.title, song.title);
+            appWidgetView.setTextViewText(R.id.artist, song.artistName);
+            appWidgetView.setTextViewText(R.id.album, song.albumName);
+        }
+
+        // Set correct drawable for pause state
+        int playPauseRes = isPlaying ? R.drawable.ic_pause_white_24dp : R.drawable.ic_play_arrow_white_24dp;
+        appWidgetView.setImageViewBitmap(R.id.button_toggle_play_pause, createBitmap(Util.getTintedVectorDrawable(service, playPauseRes, MaterialValueHelper.getSecondaryTextColor(service, true)), 1f));
+
+        // Set prev/next button drawables
+        appWidgetView.setImageViewBitmap(R.id.button_next, createBitmap(Util.getTintedVectorDrawable(service, R.drawable.ic_skip_next_white_24dp, MaterialValueHelper.getSecondaryTextColor(service, true)), 1f));
+        appWidgetView.setImageViewBitmap(R.id.button_prev, createBitmap(Util.getTintedVectorDrawable(service, R.drawable.ic_skip_previous_white_24dp, MaterialValueHelper.getSecondaryTextColor(service, true)), 1f));
+
+        // Link actions buttons to intents
+        linkButtons(service, appWidgetView);
+
+        // Load the album cover async and push the update on completion
+        final Context appContext = service.getApplicationContext();
+        final int widgetImageSize = service.getResources().getDimensionPixelSize(R.dimen.app_widget_small_image_size);
```

Comment:
This will be a constant value, so I think you could move it out of the function so it's not re-computed on every update.